src/binary/coreir.cpp

```diff
@@ -119,23 +122,37 @@ int main(int argc, char *argv[]) {
   }
 
   std::ostream* sout = &std::cout;
-  std::ofstream fout;
-  string outExt = "json";
-  string outfileName = "";
-  if (options.count("o")) {
-    outfileName = options["o"].as<string>();
-    outExt = getExt(outfileName);
-    ASSERT(outExt == "json" 
-        || outExt == "txt"
-        || outExt == "fir"
-        || outExt == "py"
-        || outExt == "smt2"
-        || outExt == "smv"
-        || outExt == "v", "Cannot support out extention: " + outExt);
-    fout.open(outfileName);
-    ASSERT(fout.is_open(),"Cannot open file: " + outfileName);
-    sout = &fout;
-  }
+  std::string outExt = "json";
+  std::string output_dir = "";
+  auto parse_outputs = [&] {
+    std::string outfile = "";
+    if (options.count("o")) {
+      outfile = options["o"].as<string>();
+      outExt = getExt(outfile);
+      ASSERT(outExt == "json"
+             || outExt == "txt"
+             || outExt == "fir"
+             || outExt == "py"
+             || outExt == "smt2"
+             || outExt == "smv"
+             || outExt == "v", "Cannot support out extention: " + outExt);
+      if (!split_files) {
+        std::unique_ptr<std::ofstream> fout(new std::ofstream(outfile));
+        ASSERT(fout->is_open(),"Cannot open file: " + outfile);
+        sout = fout.release();
+      }
+    }
+    if (split_files) {
```

Comment:
Can you add an assert here that the output file extension is verilog? and print a "NYI" otherwise.

Reply:
Done